application/controllers/Hook.php

```diff
@@ -0,0 +1,254 @@
+<?php
+defined('BASEPATH') OR exit('No direct script access allowed');
```

Comment:
unneeded

Reply:
Welcome to CodeIgniter. It's used everywhere... why exactly it is unneeded ?

Reply:
Because no code is running is, it's a class declaration.

Reply:
Makes sense